central/src/components/auth/LogIn.tsx

```diff
@@ -19,19 +19,17 @@ const LogIn: React.FC<{handleUserAuth:(isLoggedIn:boolean)=>void }> = ({handleUs
     setLoading(true);
 
     try {
-      try{
-      await Auth.signIn(email, password);
-      } catch (e) {
-        console.log(e);
-      }
-      const user = await Auth.currentAuthenticatedUser();
-      console.log(user);
-      if (user.signInUserSession.accessToken.payload["cognito:groups"].includes('Teacher_Auth')){
-        handleUserAuth(true);
-        window.location.href = "/";
+      await signIn({username: email, password: password});
+      const user = await fetchAuthSession();
+      if (user && user.tokens && user.tokens.accessToken) {
+        const groups = user.tokens.accessToken.payload["cognito:groups"];
+        if (Array.isArray(groups) && groups.includes('Teacher_Auth')) {
```

Comment:
Lets move `Teacher_Auth` into a separate variable with a meaningful name for variable name